dpnp/backend/extensions/lapack/gesv_batch.cpp

```diff
@@ -0,0 +1,326 @@
+
+        sycl::event gesv_event;
+
+        try {
+            gesv_event = mkl_lapack::gesv(
+                exec_q,
+                n,    // The order of the square matrix A
+                      // and the number of rows in matrix B (0 ≤ n).
+                nrhs, // The number of right-hand sides,
+                      // i.e., the number of columns in matrix B (0 ≤ nrhs).
+                a_batch, // Pointer to the square coefficient matrix A (n x n).
+                lda, // The leading dimension of a, must be at least max(1, n).
+                current_ipiv, // The pivot indices that define the permutation
+                              // matrix P; row i of the matrix was interchanged
+                              // with row ipiv(i), must be at least max(1, n).
+                b_batch, // Pointer to the right hand side matrix B (n x nrhs).
+                ldb,     // The leading dimension of matrix B,
+                         // must be at least max(1, n).
+                current_scratch_gesv, // Pointer to scratchpad memory to be used
+                                      // by MKL routine for storing intermediate
+                                      // results.
+                scratchpad_size, current_dep);
+        } catch (mkl_lapack::exception const &e) {
+            is_exception_caught = true;
+            info = e.info();
+
+            if (info < 0) {
+                error_msg << "Parameter number " << -info
+                          << " had an illegal value.";
+            }
+            else if (info == scratchpad_size && e.detail() != 0) {
+                error_msg << "Insufficient scratchpad size. Required size is "
+                             "at least "
+                          << e.detail();
+            }
+            else if (info > 0) {
+                T host_U;
+                exec_q
+                    .memcpy(&host_U, &a[(info - 1) * lda + info - 1], sizeof(T))
+                    .wait();
+
+                using ThresholdType = typename helper::value_type_of<T>::type;
+
+                const auto threshold =
+                    std::numeric_limits<ThresholdType>::epsilon() * 100;
+                if (std::abs(host_U) < threshold) {
+                    sycl::free(scratchpad, exec_q);
+                    sycl::free(ipiv, exec_q);
+                    throw LinAlgError(
+                        "The input coefficient matrix is singular.");
+                }
+                else {
+                    error_msg
+                        << "Unexpected MKL exception caught during gesv() "
+                           "call:\nreason: "
+                        << e.what() << "\ninfo: " << e.info();
+                }
+            }
+            else {
+                error_msg << "Unexpected MKL exception caught during gesv() "
+                             "call:\nreason: "
+                          << e.what() << "\ninfo: " << e.info();
+            }
```

Comment:
I guess that is the same between `gesv` and `gesv_batch` and so can be separated to a helper call.

Reply:
I have implemented `handle_lapack_exc` to `gesv_common_utils.hpp`.
Thanks